readthedocs/proxito/README.rst

```diff
@@ -5,6 +5,41 @@ Module in charge of serving documentation pages.
 
 Read the Docs core team members can view the `Proxito design doc <https://github.com/readthedocs/el-proxito/blob/master/docs/design/architecture.rst>`_
 
+URL parts
+---------
+
+In our code we use the following terms to refer to the different parts of the URL:
+
+url:
+   The full URL, for example ``https://docs.readthedocs.io/en/latest/index.html``.
+path:
+   The whole path from the URL, for example ``/en/latest/index.html``.
+domain:
+   The domain/subdomain without the protocol, for example ``docs.readthedocs.io``.
+language:
+   The language of the documentation, for example ``en``.
+version:
+   The version of the documentation, for example ``latest``.
+filename:
+   The name of the file being served, for example ``index.html``.
+path prefix:
+   The path prefix of the URL without version or language,
+   for a normal project this is ``/``, and for subprojects this is ``/projects/<subproject-alias>/``.
+
+.. code:: text
+
+                         URL
+   |------------------------------------------------|
+                                        path
+                              |---------------------|
+    https://docs.readthedocs.io/en/latest/index.html
+   |-------|-------------------|--|------|----------|
+    protocol         |          |     |       |
+                   domain       |     |       |
+                             language |       |
+                                    version   |
+                                           filename
```

Comment:
:100:

Reply:
This is great. Though this is varying a bit from urlparse in some ways which make sense, but we should be explicit if we don't care about the other parts (eg. `query` & `fragment`).

`urlparse("scheme://netloc/path;parameters?query#fragment")`

https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse